src/cpp/lpnamer/Candidate.cpp

```diff
@@ -1,5 +1,16 @@
 #include "Candidate.h"
 
+Candidate::Candidate(CandidateData data, const std::map<std::string, LinkProfile>& profile_map)
+{
+    _data = data;
+    _name = _data.name;
+    auto it = profile_map.find(data.link_profile);
+    if (it != profile_map.end())
+    {
+        _profile = it->second;
+    }
+}
+
```

Comment:
Maybe pass directly the profile instead of the map ?
Behavior expected if map doesnt contains profile ?